core/trino-main/src/main/java/io/trino/sql/DynamicFilters.java

```diff
@@ -402,7 +402,7 @@ public static boolean dynamicFilter(@SqlType("T") double input, @SqlType(VARCHAR
     {
         private NullableFunction() {}
 
-        private static final String NAME = "$internal$dynamic_filter_nullable_function";
```

Comment:
If I'm reading the JLS correctly, this seems to be a bug in the Eclipse compiler:

> A member (class, interface, field, or method) of a class, interface, type parameter,
or reference type, or a constructor of a class, is accessible only if (i) the class, interface, type parameter, or reference type is accessible, and (ii) the member or
constructor is declared to permit access:
> [...]
> – Otherwise, the member or constructor is declared private. Access is
permitted only when the access occurs from within the body of the top level
class or interface that encloses the declaration of the member or constructor.

and 

> The body of a class declares members (fields, methods, classes, and interfaces),
instance and static initializers, and constructors (§8.1.7).